src/docs/spec-formats-tsv.md

```diff
@@ -0,0 +1,274 @@
+
+| `match_type` value | `mapping_justification` value |
+| ------------------ | ----------------------------- |
+| Lexical            | semapv:LexicalMatching        |
+| Logical            | semapv:LogicalMatching        |
+| HumanCurated       | semapv:ManualMappingCuration  |
+| Complex            | semapv:CompositeMatching      |
+| Unspecified        | semapv:UnspecifiedMatching    |
+| SemanticSimilarity | semapv:SemanticSimilarityThresholdMatching |
+
+Any other value in the `match_type` slot MUST be treated as an error.
+
+If the set contains both `match_type` and `mapping_justification` slots, it is advised to simply ignore the former.
+
+
+#### `match_term_type` slot
+
+Initial versions of this specification defined a `match_term_type` slot on the `Mapping` class. The slot was intended to describe what was being matched. In SSSOM 0.9.1, this slot was replaced by two distinct slots called `subject_type` and `object_type` (this notably allowed for the case where the subject and the object are of a different type, something the `match_term_type` slot did not support).
+
+Upon encountering a `match_term_type` slot, implementations supporting pre-1.0 versions MUST silently transform it into a pair of `subject_type` and `object_type` slots, both slots having the same value derived from the original value using the following table:
+
+| `match_term_type` value | `subject_type` and `object_type` value |
+| ----------------------- | -------------------------------------- |
+| ConceptMatch            | skos concept                           |
+| ClassMatch              | owl class                              |
+| ObjectPropertyMatch     | owl object property                    |
+| IndividualMatch         | owl named individual                   |
+| DataPropertyMatch       | owl data property                      |
+| TermMatch               | rdfs literal                           |
+
+Any other value in the `match_term_type` slot MUST be treated as an error.
+
+If the set already contains `subject_type` and `object_type` slots, any `match_term_type` slot can be silently ignored.
+
+
+## Canonical SSSOM/TSV format
+
+This section defines a “canonical” variant of the SSSOM/TSV format, which has stricter serialisation rules. The purpose of the canonical SSSOM/TSV format is to minimise differences across SSSOM/TSV files that would be induced by small diverging behaviours between different SSSOM/TSV implementations.
+
+The rules in this section apply to SSSOM/TSV writers only. SSSOM/TSV writers SHOULD write files in the canonical format, but SSSOM/TSV readers MUST NOT reject a file solely because it does not follow the canonical rules.
+
+### General rules
+
+A canonical SSSOM/TSV writer:
+
+* MUST use line breaks made of only the U+000A character (no U+000D, and no U+000D + U+000A sequences);
+* MUST condense the slots whenever possible, as described in the [Condensation](#condensation) section.
```

Comment:
So either propagation and condensation should both be recommended, or they should both be optional. But if they are optional, then I’d question why we even bother having “propagatable slots” at the mapping set level.

If propagation is optional, we can expect that many implementations will not support it (they won’t have to -- it’s optional). Then it means that people (even those crafting their mapping sets manually) will be wary of using propagatable slots, because they will not be sure whether they will be interpreted (propagated) correctly.

So no one will ever use propagatable slots, and the entire feature will exist for nothing. Then, again, why do we bother having this feature? We might as well remove all “propagatable slots” from the `MappingSet` class.

Reply:
Now that I get the point better, I agree with this! I read this section a bit differently the first time around (like a value judgement - your Mapping set is somehow worse when it is not in normal form), but having a clearly defined _canonical_ form is obviously very valuable for many reasons.

Reply:
> like a value judgement - your Mapping set is somehow worse when it is not in normal form

That is absolutely not how the ”canonical format” should be understood, and if you have any suggestion as to how make that clearer in the spec, I’m listening.

Basically, the point of the canonical format is to avoid diffs like this:

```diff
-UBERON:0001931   skos:exactMatch   BTO:0006166   semapv:ManualMappingCuration   0.556
+UBERON:0001931   skos:exactMatch   BTO:0006166   semapv:ManualMappingCuration   0.556000
```

where the mapping has not actually changed, but the file is different all the same because the old file was written by an implementation that truncated floating values to 3 digits after the decimal point, while the new file was written by an implementation that truncates floating values to 6 digits.

Reply:
I think this is fine. The spec does not need to provide guidelines on how to "feel about the spec". I was missing the thought "canonical format --> diff" and if I would have had that in my mind like a good engineer I would not have even stumbled across this. All good.